File: src/main/java/com/ssafy/ChallenMungs/challenge/common/repository/ChallengeRepository.java
package com.ssafy.ChallenMungs.challenge.common.repository;

import com.ssafy.ChallenMungs.challenge.common.entity.Challenge;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.querydsl.QuerydslPredicateExecutor;

import java.util.List;

public interface ChallengeRepository extends JpaRepository<Challenge, Long>, QuerydslPredicateExecutor<Challenge> {
    List<Challenge> findByChallengeType(int i);
}

File: ChallenMungs/app/src/main/java/com/ssafy/challenmungs/domain/repository/ChallengeRepository.kt
package com.ssafy.challenmungs.domain.repository

import com.ssafy.challenmungs.data.remote.Resource
import com.ssafy.challenmungs.domain.entity.challenge.Challenge

interface ChallengeRepository {

    suspend fun getChallengeList(
        type: Int,
        searchValue: String?,
        lat: Double,
        lng: Double
    ): Resource<List<Challenge>>
}